tests/pyspark_delta/test_pyspark_writer.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+import chispa
+import pyspark
+from delta import *
+import shutil
+import os
+
+builder = (
+    pyspark.sql.SparkSession.builder.appName("MyApp")
+    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+    .config(
+        "spark.sql.catalog.spark_catalog",
+        "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+    )
+)
+
+spark = configure_spark_with_delta_pip(builder).getOrCreate()
+
+def test_reference_table_1_append():
+    os.makedirs("tmp")
+
+    # copy over existing reference table
+    shutil.copytree("out/tables/generated/reference_table_1/delta", "tmp/delta")
+    df = spark.read.format("delta").load("tmp/delta")
+    # +------+------+-------+                                                         
+    # |letter|number|a_float|
+    # +------+------+-------+
+    # |     a|     1|    1.1|
+    # |     b|     2|    2.2|
+    # |     c|     3|    3.3|
+    # |     d|     4|    4.4|
+    # |     e|     5|    5.5|
+    # +------+------+-------+
+    
+    # append data to the reference table
+    rdd = spark.sparkContext.parallelize([("z", 9, 9.9)])
+    df = rdd.toDF(["letter", "number", "a_float"])
+    df.write.format("delta").mode("append").save("tmp/delta")
```

Comment:
This will be one of the main challenges: how can we describe these operations in an engine agnostic way? There's two possible levels to do this at:

 * A **machine-readable description** that could be automatically translated. This would be *nice*, but I'm not confident it would be feasible.
 * A **human-readable description** that would have to be hand-written by implementors. I *think* this would be sufficient since the semantics of it are going to be checked by the `expected` result.

Reply:
Yep, I vote for the "human-readable" approach as the first pass and then see if we can make something more clever down the road when we know the problem a bit better.

Reply:
+1 for human-readable as a first pass.